pkg/config/virtio.go

```diff
@@ -663,6 +679,76 @@ func (dev *RosettaShare) FromOptions(options []option) error {
 	return nil
 }
 
+func networkBlockDeviceNewEmpty() *NetworkBlockDevice {
+	return &NetworkBlockDevice{
+		VirtioBlk: VirtioBlk{
+			StorageConfig: StorageConfig{
+				DevName: "nbd",
+			},
+			DeviceIdentifier: "",
+		},
+		Timeout:             time.Duration(15000 * time.Millisecond), // set a default timeout to 15s
+		SynchronizationMode: SynchronizationFullMode,                 // default mode to full
+	}
+}
+
+// NetworkBlockDeviceNew creates a new disk by connecting to a remote Network Block Device (NBD) server.
+// The provided uri must be in the format <scheme>://<address>/<export-name>
+// where scheme could have any of these value: nbd, nbds, nbd+unix and nbds+unix.
+// More info can be found at https://github.com/NetworkBlockDevice/nbd/blob/master/doc/uri.md
+// This allows the virtual machine to access and use the remote storage as if it were a local disk.
+func NetworkBlockDeviceNew(uri string, timeout uint32, synchronization NBDSynchronizationMode) (*NetworkBlockDevice, error) {
+	nbd := networkBlockDeviceNewEmpty()
+	nbd.URI = uri
+	nbd.Timeout = time.Duration(timeout) * time.Millisecond
+	nbd.SynchronizationMode = synchronization
+
+	return nbd, nil
+}
+
+func (nbd *NetworkBlockDevice) ToCmdLine() ([]string, error) {
+	cmdLine, err := nbd.VirtioBlk.ToCmdLine()
+	if err != nil {
+		return []string{}, err
+	}
+
```

Comment:
I'd add the same check as in `VirtioBlk.ToCmdLine()`:
```
	if len(cmdLine) != 2 {
		return []string{}, fmt.Errorf("unexpected storage config commandline")
	}
```

Reply:
I think I didn't add it bc in the row above it calls `cmdLine, err := nbd.VirtioBlk.ToCmdLine()` so the check is already performed in the `VirtioBlk.ToCmdLine()`. Do you prefer to have it duplicated?

Reply:
In my opinion it is better to have it here as well. If one day we refactor this code and it no longer calls `VirtioBlk.ToCmdLine()` we will lose that check without noticing.